src/lib/djerba/plugins/hrd/plot.R

```diff
@@ -0,0 +1,102 @@
+#! /usr/bin/env Rscript
+
+library(dplyr)
+library(ggplot2)
+library(optparse)
+library(scales)
+library(cowplot)
+
+option_list = list(
+  make_option(c("-d", "--dir"), type="character", default=NULL, help="Input report directory path", metavar="character")
+)
+
+# get options
+opt_parser <- OptionParser(option_list=option_list, add_help_option=FALSE)
+opt <- parse_args(opt_parser)
+work_dir <- opt$dir
+
+hrd_color <- c("#242424" ,"#d2d2d2","#999999",  "#e8e8e8" ,"#000000")
+options(bitmapType='cairo')
+
+cutoff_low = 0.7
+cutoff_high = cutoff_low
+
+boot <- read.table(paste(work_dir, 'hrd.tmp.txt', sep='/'),header=FALSE)
+
+names(boot) <- c("var","q1","median_value","q3")
+boot$Sample <- "Sample"
+
+hrd_median <- as.numeric(unique(boot$median_value[boot$var == "Probability.w"])) 
+intercept  <- boot$median_value[boot$var == "intercept.w"]
+
+weights_df <- boot[boot$var %in% c("del.mh.prop.w","SNV3.w","SV3.w","SV5.w","hrd.w"),]
+
+#equation [4] in Davies et al. 2017
+weights_df$probability <- 1 / ( 1 + exp(-(intercept + weights_df$median_value)))
+
+weights_df$var_long[weights_df$var == "SV5.w"] <- "Large Deletions"
+weights_df$var_long[weights_df$var == "SV3.w"] <- "Tandem Duplications"
+weights_df$var_long[weights_df$var == "SNV3.w"] <- "COSMIC SBS3"
+weights_df$var_long[weights_df$var == "hrd.w"] <- "LOH"
+weights_df$var_long[weights_df$var == "del.mh.prop.w"] <- "Microhomologous Deletions"
+weights_df$var_longer <- paste0(weights_df$var_long,": ", round(weights_df$probability,2))
+
+weights_df <- weights_df[order(weights_df$var_long,decreasing = T),]
+
+weights_df$probability_cum <- cumsum(weights_df$probability)
+weights_df$probability_cum_half <- (weights_df$probability / 2) + c(0,weights_df$probability_cum[-5])
+
+
+out_path <- paste(work_dir, 'hrd.svg', sep='/')
+
+svg(out_path, width = 8, height = 1.5, bg = "transparent")
+
+
+  print(
+    ggplot(weights_df ,aes(x="Sample",y=as.numeric(probability))) + 
+      
+      geom_bar(aes(fill=var_longer),stat ="identity",width=0.5, color = "white") + 
+      
+      annotate(x = 0, xend=2, y=cutoff_low, yend=cutoff_low, geom="segment",colour = "gray",linetype="dashed") +
+      annotate(geom="text",x = 0,y=cutoff_low/2,color="gray30",label="HR-P", hjust = 0.5, vjust = 2,size=4) +
+      
+      annotate(geom="text",x = 0,y=(cutoff_high + max(hrd_median, 0.40))/2, color="gray30",label="HR-D", hjust = 0.5, vjust = 2,size=4) +
+      
+   #   geom_point(aes(y=as.numeric(probability_cum_half),x=c(1,1,1,1,1))) +
```

Comment:
Is this code needed as an example or for future reference? If so, better add some comments to explain why. If not, can delete it. :fire: